### PR TITLE
Feature/emp 3542 migrate global x bus component

### PR DESCRIPTION
**Pull request by albertjcuac**
# Pull request template
 
For the vue 3 migration we need to migrate components to 2.7. 
Snippet callbacks tests have been commented because they have not been migrated yet and they use globalXbus , so we need to avoid them failling for this PR and fix them in the Snippet Callbacks migration task.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link: [EMP-3542](https://searchbroker.atlassian.net/browse/EMP-3542)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

testing it still works following the docs usage: [link](https://docs.empathy.co/develop-empathy-platform/ui-reference/components/base-components/x-components.global-x-bus.html)

[EMP-3542]: https://searchbroker.atlassian.net/browse/EMP-3542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by albertjcuac** [comment]
Snippet callbacks tests have been commented because they have not been migrated yet and they use globalXbus , so we need to avoid them failling for this PR and fix them in the Snippet Callbacks migration task.

File: packages/x-components/src/components/__tests__/snippet-callbacks.spec.ts [hunk 1]
```diff
@@ -21,71 +23,89 @@ function renderSnippetCallbacks({
 }
 
 describe('testing SnippetCallbacks component', () => {
```
Comment (CachedaCodes) [inline packages/x-components/src/components/__tests__/snippet-callbacks.spec.ts:25]: Revert the changes here and skip these tests, they'll be fixed in the snippet callbacks task

```suggestion
describe.skip('testing SnippetCallbacks component', () => {
```

File: packages/x-components/src/components/__tests__/global-x-bus.spec.ts [hunk 2]
```diff
@@ -1,36 +1,41 @@
-import { mount, Wrapper } from '@vue/test-utils';
+import { mount } from '@vue/test-utils';
 import Vue, { ComponentOptions } from 'vue';
 import { installNewXPlugin } from '../../__tests__/utils';
 import GlobalXBus from '../global-x-bus.vue';
+import { bus } from '../../plugins/x-bus';
 
-function renderGlobalXBus({ listeners = {} }: RenderGlobalXBusOptions = {}): RenderGlobalXBusAPI {
+function renderGlobalXBus({ listeners = {} }: RenderGlobalXBusOptions = {}): void {
   const [, localVue] = installNewXPlugin();
```
Comment (CachedaCodes) [inline packages/x-components/src/components/__tests__/global-x-bus.spec.ts:8]: The x plugin is getting deprecated, so don't use `installNewXPlugin` (not even sure if you need it for this test)

File: .eslintrc.js [hunk 3]
```diff
@@ -7,7 +7,8 @@ module.exports = {
     '**/jest.setup.ts',
     '**/dist',
     'packages/search-adapter',
-    'packages/eslint-plugin-x'
+    'packages/eslint-plugin-x',
+    'packages/x-components/src/components/__tests__/snippet-callbacks.spec.ts'
```
Comment (CachedaCodes) [inline .eslintrc.js:11]:
```suggestion
```

File: packages/x-components/src/components/__tests__/global-x-bus.spec.ts
Comment (CachedaCodes) [inline packages/x-components/src/components/__tests__/global-x-bus.spec.ts:1]: Add a test checking that the subscriptions are unsubscribed when the GlobalXBus component is unmounted.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by annacv — APPROVED** [review]
🎉

**Pull request merged**